src/System.Runtime/tests/Performance/Perf.Double.cs

```diff
@@ -12,466 +14,121 @@ namespace System.Tests
-        [InlineData("E", Math.PI)]
-        [InlineData("E", Math.E)]
-        [InlineData("E", double.MaxValue)]
-        [InlineData("E", double.MinValue)]
-        [InlineData("E", double.NaN)]
-        [InlineData("E", double.PositiveInfinity)]
-        [InlineData("E", double.NegativeInfinity)]
-        [InlineData("E", 2.2250738585072009E-308)]
-        [InlineData("E", -2.2250738585072009E-308)]
-        [InlineData("E", 1.1125369292536E-308)]
-        [InlineData("E", -0.0)]
-        [InlineData("E", 0.0)]
-        public void ToStringWithFormat(string format, double number)
+        public static IEnumerable<object[]> ToStringWithFormat_TestData()
+        {
+            string[] formats =
+            {
+                "R",
+                "G",
+                "G17",
+                "E",
+                "F"
+            };
+
+            double[] normalTestValues =
+            {
+                0.0,
+                250.0,
+            };
+
+            double[] edgeTestValues =
+            {
+                double.MaxValue,
+                double.MinValue,
+                double.Epsilon,
+            };
+
+            foreach (string format in formats)
+            {
+                foreach (double testValue in normalTestValues)
+                {
+                    yield return new object[] { format, testValue, 2_000_000 };
+                }
+
+                foreach (double testValue in edgeTestValues)
+                {
+                    yield return new object[] { format, testValue, 100_000 };
+                }
+            }
+
+            yield return new object[] { "G", double.PositiveInfinity, 20_000_000 };
+            yield return new object[] { "G", double.NaN, 20_000_000 };
+        }
+
+        [Benchmark]
+        [MemberData(nameof(ToStringWithFormat_TestData))]
+        public void ToStringWithFormat(string format, double number, int innerIterations)
         {
             foreach (var iteration in Benchmark.Iterations)
             {
                 using (iteration.StartMeasurement())
                 {
-                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    for (int i = 0; i < innerIterations; i++)
                     {
-                        number.ToString(format);
+                        _string = number.ToString(format);
                     }
                 }
             }
         }
 
-        [Benchmark]
+        [Benchmark(InnerIterationCount = 4_000_000)]
         public static void Decimal_ToString()
         {
             decimal decimalNum = new decimal(dValue);
             foreach (var iteration in Benchmark.Iterations)
+            {
                 using (iteration.StartMeasurement())
-                    decimalNum.ToString();
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        decimalNum.ToString();
```

Comment:
This should write the return value into a volatile static or member variable (there are no observable side-effects, and the entire operation can be optimized out by the JIT).